girder/models/user.py

```diff
@@ -244,26 +249,43 @@ def createUser(self, login, password, firstName, lastName, email,
             'groupInvites': []
         }
 
-        self.setPassword(user, password)
+        self.setPassword(user, password, save=False)
+
+        self.setPublic(user, public, save=False)
+
+        # An id is required in order to grant the user access on itself, but
+        # since the user isn't fully created yet, it should not be saved
+        user['_id'] = ObjectId()
```

Comment:
Anything I can think of is pretty convoluted. I'm tempted to just not try and fix this at all since the database state would be really easy for an admin to fix via the UI in the rare event that the initial query succeeds and the second fails.